ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/ScheduledBlockFetchServer.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Consensus.PeerSimulator.ScheduledBlockFetchServer (
+    BlockFetch (..)
+  , ScheduledBlockFetchServer (..)
+  , runScheduledBlockFetchServer
+  ) where
+
+import           Control.Tracer
+import           Data.List (intercalate)
+import           Ouroboros.Consensus.Block (Point)
+import           Ouroboros.Consensus.Util.Condense (Condense)
+import           Ouroboros.Consensus.Util.IOLike (IOLike, MonadSTM (STM))
+import           Ouroboros.Network.BlockFetch.ClientState (ChainRange)
+import           Ouroboros.Network.Protocol.BlockFetch.Server
+import           Test.Consensus.PeerSimulator.ScheduledServer
+                     (ScheduledServer (..), awaitOnlineState, runHandler)
+import           Test.Consensus.PeerSimulator.Trace
+import           Test.Util.TestBlock (TestBlock)
+
+data SendBlocks =
```

Comment:
Funnily enough, this data type is unused. I removed it in e62e8521801ff157694af31139c8776ea5adbb01. It will be reintroduced in https://github.com/IntersectMBO/ouroboros-consensus/pull/840 when it is actually useful.